File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.java
class Solution:
    def minimumDeletions(self, nums: List[int]) -> int:
        n = len(nums)
        left = 0
        right = 0

        for i in range(1, n):
            if nums[i] < nums[left]:
                left = i

            if nums[i] > nums[right]:
                right = i

        if left < right:
            left, right = right, left

        ans = n

        for i in range(n + 1):
            extra = 0

            if right >= i:
                extra = n - right
            elif left >= i:
                extra = n - left

            ans = min(ans, i + extra)

        return ans

File: 2091-removing-minimum-and-maximum-from-array/2091-removing-minimum-and-maximum-from-array.cpp
class Solution {
public:
    int minimumDeletions(vector<int>& nums) {
        int n = nums.size();
        int left = 0;
        int right = 0;

        for (int i = 1; i < n; i++) {
            if (nums[i] < nums[left])
                left = i;

            if (nums[i] > nums[right])
                right = i;
        }

        if (left < right)
            swap(left, right);

        int ans = n;

        for (int i = 0; i <= n; i++) {
            int extra = 0;

            if (right >= i)
                extra = n - right;
            else if (left >= i)
                extra = n - left;

            ans = min(ans, i + extra);
        }

        return ans;
    }
};